File: code/stable/projectile/projectile_c_p_nol_b_u_v_d/src/csharp/Control.cs
/** \file Control.cs
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Controls the flow of the program
*/
public class Control {
    
    /** \brief Controls the flow of the program
        \param args List of command-line arguments
    */
    public static void Main(string[] args) {
        string filename = args[0];
        double g_vect = 9.8;
        double epsilon = 2.0e-2;
        InputParameters inParams = new InputParameters(filename);
        double t_flight = Calculations.func_t_flight(inParams, g_vect);
        double p_land = Calculations.func_p_land(inParams, g_vect);
        double d_offset = Calculations.func_d_offset(inParams, p_land);
        string s = Calculations.func_s(inParams, epsilon, d_offset);
        OutputFormat.write_output(s, d_offset, t_flight);
    }
}

File: code/stable/projectile/projectile_c_p_nol_b_u_v_d/src/cpp/Control.cpp
/** \file Control.cpp
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Controls the flow of the program
*/
#include <string>

#include "Calculations.hpp"
#include "InputParameters.hpp"
#include "OutputFormat.hpp"

using std::string;

/** \brief Controls the flow of the program
    \param argc Number of command-line arguments
    \param argv List of command-line arguments
    \return exit code
*/
int main(int argc, const char *argv[]) {
    string filename = argv[1];
    double g = 9.8;
    double epsilon = 2.0e-2;
    InputParameters inParams = InputParameters(filename);
    double t_flight = func_t_flight(inParams, g);
    double p_land = func_p_land(inParams, g);
    double d_offset = func_d_offset(inParams, p_land);
    string s = func_s(inParams, epsilon, d_offset);
    write_output(s, d_offset, t_flight);
    
    return 0;
}
